File: core-java/src/main/java/hello/core/discount/FixDiscountPolicy.java
package hello.core.discount;

import hello.core.member.Grade;
import hello.core.member.Member;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.stereotype.Component;

@Component
//@Qualifier("fixDiscountPolicy")
public class FixDiscountPolicy implements DiscountPolicy{

    private final int discountFixAmount = 1000;
    @Override
    public int discount(Member member, int price) {
        // enum 은 == 으로 비교
        if (member.getGrade() == Grade.VIP) {
            return discountFixAmount;
        }
        return 0;
    }
}

File: core-kotlin/src/main/kotlin/hello/core/discount/FixDiscountPolicy.kt
package hello.core.discount

import hello.core.member.Grade
import hello.core.member.Member

class FixDiscountPolicy : DiscountPolicy {

    val discountFixAmount: Int = 1000
    override fun discount(member: Member, price: Int): Int {
        if (member.garde === Grade.VIP) {
            return discountFixAmount
        }
        return 0
    }
}